thirdparty/patches/opencensus-cpp-start-harvest-thread.patch

```diff
@@ -0,0 +1,45 @@
+diff --git opencensus/stats/internal/delta_producer.cc opencensus/stats/internal/delta_producer.cc
+--- opencensus/stats/internal/delta_producer.cc
++++ opencensus/stats/internal/delta_producer.cc
+@@ -128,10 +128,8 @@ void DeltaProducer::Flush() {
+   ConsumeLastDelta();
+ }
+ 
+-DeltaProducer::DeltaProducer()
+-    : harvester_thread_(&DeltaProducer::RunHarvesterLoop, this) {
+-  absl::MutexLock l(&mu_);
+-  thread_started_ = true;
++DeltaProducer::DeltaProducer() {
++  StartHarvestLoopIfNeeded();
```

Comment:
Previously, when `DeltaProducer.Shutdown` is called, it is blocked until it flushes the remaining metrics. But if I remove Shutdown, this logic was removed which caused some test failures. Instead of removing shutdown, I made it restart the thread via `StartHarvestLoopIfNeeded `